bot/exts/evergreen/avatar_modification/avatar_modify.py

```diff
@@ -308,7 +313,57 @@ async def spookyavatar(self, ctx: commands.Context, member: discord.Member = Non
 
             await ctx.send(file=file, embed=embed)
 
+    @avatar_modify.command(name="mosaic", root_aliases=("mosaic",))
+    async def mosaic_command(self, ctx: commands.Context, squares: int = 16) -> None:
+        """Splits your avatar into x squares, randomizes them and stitches them back into a new image!"""
+        async with ctx.typing():
+            member = await self._fetch_member(ctx.author.id)
+            if not member:
+                await ctx.send(f"{Emojis.cross_mark} Could not get member info.")
+                return
+
+            if squares < 1:
+                raise commands.BadArgument("Squares must be a positive number")
+
+            if not math.sqrt(squares).is_integer():
+                raise commands.BadArgument("Squares must be a perfect square")
+
+            if squares > MAX_SQUARES:
+                raise commands.BadArgument(f"Number of squares cannot be higher than {MAX_SQUARES:,}.")
+
+            file_name = file_safe_name("mosaic_avatar", ctx.author.display_name)
+
+            img_bytes = await member.avatar_url.read()
+
+            file = await in_executor(
+                PfpEffects.mosaic_effect,
+                img_bytes,
+                squares,
+                file_name
+            )
+
+            if squares == 1:
+                title = "Hooh... that was a lot of work"
+                description = "I present to you... Yourself!"
+            elif squares == MAX_SQUARES:
+                title = "Testing the limits I see..."
+                description = "What a masterpiece. :star:"
+            else:
+                title = "Your mosaic avatar"
+                description = "Here is your avatar. I think it looks a bit *puzzling*"
+
+            embed = discord.Embed(
+                title=title,
+                description=description,
+                colour=Colours.blue
```

Comment:
Nitpicking, but am not a fan of this color, we can probably keep it the default color (even `8bitify` uses that).